packages/js/src/utils/id-token.ts

```diff
@@ -0,0 +1,68 @@
+import * as s from 'superstruct';
+
+const fullfillBase64 = (input: string) => {
+  if (input.length === 2) {
+    return `${input}==`;
+  }
+
+  if (input.length === 3) {
+    return `${input}=`;
+  }
+
+  return input;
+};
+
+/**
+ * @link [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+ */
+const IdTokenClaimsSchema = s.type({
+  iss: s.string(),
+  sub: s.string(),
+  aud: s.string(),
+  exp: s.number(),
+  iat: s.number(),
+  auth_time: s.optional(s.number()),
+  nonce: s.optional(s.string()),
+  acr: s.optional(s.string()),
+  amr: s.optional(s.array(s.string())),
+  azp: s.optional(s.string()),
+  at_hash: s.optional(s.string()),
+  c_hash: s.optional(s.string()),
+});
+
+export type IdTokenClaims = s.Infer<typeof IdTokenClaimsSchema>;
+
+/**
+ * Decode ID Token from JWT, without verifying.
+ *
+ * Verifying JWT requires fetching public key first,
+ * this can not be done in a sync function,
+ * in some cases, verifying is not necessary.
+ * @param token JWT string
+ * @returns IDToken combined with JWT Claims.
+ */
+export const decodeIdToken = (token: string): IdTokenClaims => {
+  const payloadPart = token.split('.')[1];
+
+  if (!payloadPart) {
+    throw new Error('invalid token');
+  }
+
+  const payloadString = payloadPart.replace(/-/g, '+').replace(/_/g, '/');
+  const json = decodeURIComponent(
+    escape(Buffer.from(fullfillBase64(payloadString), 'base64').toString())
+  );
+
```

Comment:
why do we need decode JWT manually if we already imported `jose`? please see
https://github.com/panva/jose/blob/main/docs/functions/jwt_verify.jwtVerify.md#readme

Reply:
`jwtVerify` require key.
`decodeIdToken` have only one parameter `idToken` and it cannot get the key.

Reference: https://www.notion.so/silverhand/decodeIdToken-7b3b739c397a431882a2e41bac8782f1

Reply:
Should use `UnsecuredJWT`, which doesn't require key, instead.

Reference: https://github.com/panva/jose/blob/main/docs/classes/jwt_unsecured.UnsecuredJWT.md#readme

Reply:
`UnsecuredJWT` doesn't work here since 

> The UnsecuredJWT class is a utility for dealing with { "alg": "none" } Unsecured JWTs.

Reply:
<img width="774" alt="image" src="https://user-images.githubusercontent.com/36393111/150075579-aa0b4cb1-b2aa-4a4f-89cd-1a8f42d11e51.png"><img width="794" alt="image" src="https://user-images.githubusercontent.com/36393111/150075469-d49681cf-f0f2-4114-8420-89cdaadf5331.png">